### PR TITLE
Update Go version to 1.22

### PR DESCRIPTION
**Pull request by JoshVanL**
Chore

Closes #3372

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -134,6 +132,19 @@ require (
...
+	github.com/chenzhuoyu/base64x v0.0.0-20230717121745-296ad89f973d // indirect
+	github.com/cloudwego/configmanager v0.2.0 // indirect
+	github.com/cloudwego/dynamicgo v0.2.0 // indirect
+	github.com/cloudwego/localsession v0.0.2 // indirect
+	github.com/fatih/structtag v1.2.0 // indirect
+	github.com/iancoleman/strcase v0.2.0 // indirect
+	github.com/klauspost/cpuid/v2 v2.2.4 // indirect
+	github.com/modern-go/gls v0.0.0-20220109145502-612d0167dce5 // indirect
+	github.com/twitchyliquid64/golang-asm v0.15.1 // indirect
+)
```
Comment (ItalyPaleAle) [inline go.mod:146]: Nit: please make sure there's a single block for all indirect deps, to keep them sorted

File: go.mod [hunk 2]
```diff
@@ -51,7 +49,7 @@ require (
 	github.com/cinience/go_rocketmq v0.0.2
 	github.com/cloudevents/sdk-go/binding/format/protobuf/v2 v2.14.0
 	github.com/cloudevents/sdk-go/v2 v2.14.0
-	github.com/cloudwego/kitex v0.5.0
+	github.com/cloudwego/kitex v0.9.0
```
Comment (ItalyPaleAle) [inline go.mod:52]: Is there anything we can do to confirm this works? Between 0.5.0 and 0.9.0 there has been [1 year worth of releases](https://github.com/cloudwego/kitex/releases). I don't believe we have automated tests for this today, so is there anything we can do to validate it works?

**Comment by berndverst** [comment]
Look at open PRs please.

Dupe of https://github.com/dapr/components-contrib/pull/3372

**Comment by berndverst** [comment]
We should use 1.21.8 in release branch.

**Pull request closed without merging**